mysite/announcements/tasks.py

```diff
@@ -0,0 +1,30 @@
+from __future__ import absolute_import, unicode_literals
+
+from celery import task
+from django.utils import timezone
+
+from .models import Announcements
+
+
+@task()
+def check_scheduled_announcements():
+    scheduled_announcement = Announcements.objects.filter(date_time_to_publish__lte=timezone.now())\
+        .filter(sent_at=None)\
+        .prefetch_related('groups__user_set')
+
+    for announcement in scheduled_announcement:
+        group_list = announcement.groups.all()
+        user_list = []
+
+        for group in group_list:
+            users = group.user_set.all()
+            for u in users:
+                user_list.append(u)
+
+        user_list = list(set(user_list))
+
+        for u in user_list:
+            print("{} {} {}".format(u, announcement.title, announcement.message))
+
+    scheduled_announcement.update(sent_at=timezone.now())
```

Comment:
This can be again source of a bug.

How?
Schrduled announcement is a queryset, and now querysets ate evaluated lazily.
So, it might happen that the time when we were looping over announcement on line 15 the queryset evaludated to something while here it evulated to something else.

Reply:
I would add the result of queryset in scheduled_announcement in a list and update that announcement is sent.  Then I'll use that list to look for announcement and whom it needs to be sent.

Reply:
Can you please share a dummy code snippet on how you plan to ` I'll use that list to look for announcement and whom it needs to be sent.`

Reply:
scheduled_announcement = Announcements.objects.filter(date_time_to_publish__lte=timezone.now())\
        .filter(sent_at=None)\
        .prefetch_related('groups__user_set')
my_list = list(scheduled_announcement)

I can access announcement from my_list